packages/ui/src/components/Layout/LayoutPage.tsx

```diff
@@ -10,8 +10,38 @@ import { ThemeScheme } from './Types'
 
 export const PageLayoutContent = ({ children }: { children: ReactNode }) => {
 	const prefix = useClassNamePrefix()
+	const [isOverflowing, setIsOverflowing] = useState(false)
+	const contentRef = useRef<HTMLDivElement>(null)
 
-	return <div className={`${prefix}layout-page-content`}>
+	useEffect(() => {
+		const contentRefCopy = contentRef.current
+		const listener = () => {
+			const offsetWidth = Math.floor(contentRefCopy?.offsetWidth ?? 0)
+			const scrollWidth = Math.floor(contentRefCopy?.scrollWidth ?? 0)
+			const scrollLeft = Math.floor(contentRefCopy?.scrollLeft ?? 0)
+			const scrollRight = scrollWidth - offsetWidth - scrollLeft
+
+			const nextIsOverflowing = offsetWidth < scrollWidth && scrollRight > 1
+
+			if (isOverflowing !== nextIsOverflowing) {
+				setIsOverflowing(nextIsOverflowing)
```

Comment:
call setIsOverflowing without if-ing

Reply:
Sure? Wouldn't it cause unnecessary re-renders and re-draw?

Reply:
yes, sure. if you pass same value, react won't trigger rerender. 
also, after you remove it from useEffect dependencies, you will avoid uneccessary invocation of useEffect

ps: you can also remove contentRef from dependencies, because it is referentially stable